metamask-predicate/src/main.sw

```diff
@@ -0,0 +1,13 @@
+predicate;
+
+use std::{b512::B512, constants::ZERO_B256, ecr::ec_recover_address, inputs::input_predicate_data};
+
+fn main(msg_hash: b256, signature: B512, expected_public_key: b256) -> bool {
+    let res = ec_recover_address(signature, msg_hash);
+    if res.is_ok() {
+        if res.unwrap().value == expected_public_key {
+            return true;
+        }
+    }
```

Comment:
We also need to validate that `msg_hash` actually corresponds in some meaningful way to the transaction being signed. Otherwise I could sign any message, e.g. "123abc" and use that to unlock the predicate.

Based on [ERC-191](https://eips.ethereum.org/EIPS/eip-191), the eth_sign RPC will produce a message using the following format:
`keccak256("\x19Ethereum Signed Message:\n" + len(message) + message))`

So the predicate should use this as the expected format to compare the user provided `msg_hash` against.

E.g. 
```
let expected_msg = keccak("\x19Ethereum Signed Message:\n" + 32 + tx_id())
if msg_hash != expected_msg {
    return false
}
```